WCF/Shared/Implementation/ClientTelemetryChannelBase.cs

```diff
@@ -0,0 +1,361 @@
+                telemetry.Name = RemoteAddress.Uri.ToString();
+                telemetry.Data = ChannelManager.OperationMap.ContractType.Name;
+                return telemetry;
+            } catch ( Exception ex )
+            {
+                WcfClientEventSource.Log.ClientTelemetryError(method, ex.ToString());
+                return null;
+            }
+        }
+        private void StopOpenTelemetry(DependencyTelemetry telemetry, Exception error, String method)
+        {
+            if ( telemetry == null )
+            {
+                return;
+            }
+            try
+            {
+                telemetry.Success = error == null;
+                telemetry.Stop();
+                ChannelManager.TelemetryClient.TrackDependency(telemetry);
+            } catch ( Exception ex )
+            {
+                WcfClientEventSource.Log.ClientTelemetryError(method, ex.ToString());
+            }
+        }
+
+        protected DependencyTelemetry StartSendTelemetry(Message request, String method)
+        {
+            var soapAction = request.Headers.Action;
+            ClientOperation operation;
+            if ( !ChannelManager.OperationMap.TryLookupByAction(soapAction, out operation) )
+            {
+                return null;
+            }
+
+            try
+            {
+                var telemetry = new DependencyTelemetry();
+                ChannelManager.TelemetryClient.Initialize(telemetry);
+                telemetry.Start();
+                telemetry.Type = DependencyConstants.WcfClientCall;
+                telemetry.Target = RemoteAddress.Uri.Host;
+                telemetry.Name = RemoteAddress.Uri.ToString();
+                telemetry.Data = ChannelManager.OperationMap.ContractType.Name + "." + operation.Name;
+                telemetry.Properties[DependencyConstants.SoapActionProperty] = soapAction;
+                if ( operation.IsOneWay )
+                {
+                    telemetry.Properties[DependencyConstants.IsOneWayProperty] = Boolean.TrueString;
+                }
+                SetCorrelationHeaders(telemetry, request);
+                return telemetry;
+            } catch ( Exception ex )
+            {
+                WcfClientEventSource.Log.ClientTelemetryError(method, ex.ToString());
+                return null;
+            }
+        }
+        protected void StopSendTelemetry(DependencyTelemetry telemetry, Message response, Exception error, String method)
+        {
+            if ( telemetry == null )
+            {
+                return;
+            }
+            try
+            {
+                if ( error != null )
+                {
+                    telemetry.Success = false;
+                    telemetry.ResultCode = error.ToResultCode();
+                }
+                if ( response != null && response.IsFault )
+                {
+                    telemetry.Success = false;
+                    telemetry.ResultCode = "SoapFault";
```

Comment:
this is an excellent use of ResultCode! We have similar issue opened in SDK to track DNS error as a separate `ResultCode`. Thank you